File: Leetcode/Daily/23-05-2024/prob.go
func f(ind int,nums []int,k int, mp map[int]int) int{
    if len(nums)==ind{
        return 1
    }
    t := 0
    if mp[nums[ind]-k]==0 && mp[nums[ind]+k]==0{
        mp[nums[ind]]++
        t = f(ind+1,nums,k,mp)
        mp[nums[ind]]--
    }

    nt := f(ind+1,nums,k,mp)
    return t+nt
}


func beautifulSubsets(nums []int, k int) int {
    mp := make(map[int]int)
    cnt := f(0,nums,k,mp)
    return cnt-1
}

File: Leetcode/Daily/23-05-2024/prob.cpp
class Solution {
public:

    int f(int ind,vector<int>&nums,int k,unordered_map<int,int>&mp){
        if(ind==nums.size())return 1;
        int t = 0;

        
        if(!mp[nums[ind]+k] and !mp[nums[ind]-k]){
            mp[nums[ind]]++;
            t = f(ind+1,nums,k,mp);
            mp[nums[ind]]--;
        }
    
        int nt = f(ind+1,nums,k,mp);

        return t + nt;
    }

    int beautifulSubsets(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        int cnt = f(0,nums,k,mp);

        return cnt-1;
    }
};